GitUI/CommandsDialogs/FormDiff.cs

```diff
@@ -240,14 +249,47 @@ private void btnAnotherHeadCommit_Click(object sender, EventArgs e)
             PickAnotherCommit(_headRevision, ref _headCommitDisplayStr, ref _headRevision);
         }
 
-        private void openWithDifftoolToolStripMenuItem_DropDownOpening(object sender, System.ComponentModel.CancelEventArgs e)
+        private void diffContextToolStripMenuItem_Opening(object sender, System.ComponentModel.CancelEventArgs e)
         {
-            aLocalToolStripMenuItem.Enabled = _baseRevision != null && _baseRevision.Guid != GitRevision.UnstagedGuid && !Module.IsBareRepository();
-            bLocalToolStripMenuItem.Enabled = _headRevision != null && _headRevision.Guid != GitRevision.UnstagedGuid && !Module.IsBareRepository();
-            parentOfALocalToolStripMenuItem.Enabled = parentOfBLocalToolStripMenuItem.Enabled = !Module.IsBareRepository();
-
+            bool isAnyTracked = DiffFiles.SelectedItems.Any(item => item.IsTracked);
             bool isExactlyOneItemSelected = DiffFiles.SelectedItems.Count() == 1;
-            blameToolStripMenuItem.Visible = isExactlyOneItemSelected && !(DiffFiles.SelectedItem.IsSubmodule || _baseRevision.IsArtificial);
+
+            openWithDifftoolToolStripMenuItem.Enabled = isAnyTracked;
+            fileHistoryDiffToolstripMenuItem.Enabled = isAnyTracked && isExactlyOneItemSelected;
+            blameToolStripMenuItem.Enabled = fileHistoryDiffToolstripMenuItem.Enabled && !DiffFiles.SelectedItem.IsSubmodule;
+        }
+
+        private ContextMenuDiffToolInfo GetContextMenuDiffToolInfo()
+        {
+            bool firstIsParent = _revisionDiffController.IsFirstParent(DiffFiles.Revision.ParentGuids, DiffFiles.SelectedItemParents.Select(i => i.Guid));
+            bool localExists = _revisionDiffController.LocalExists(DiffFiles.SelectedItemsWithParent, _fullPathResolver);
```

Comment:
this indicates an incorrect use of `IFullPathResolver` - behaviors must be injected via constructors and not passed as parameters

Reply:
I see two two solutions for the relative simple LocalExists() function:
 * Add LocalExists() to IFullPathResolver instead
 *  Duplicate LocalExists() where it is used (two or three modules). A separate interface/class for this seem to be overkill.

Reply:
> Add LocalExists() to IFullPathResolver instead

:+1:

Reply:
I duplicated LocalExists() as it deals with GitItemStatusWithParent.
Probably better to move other similar to a separate class (but FullPathResolver.Resolve() will be duplicated instead).